src/function/table/system/attach_duckdb.cpp

```diff
@@ -0,0 +1,241 @@
+		names.emplace_back(result->tablerelation->names[i]);
+	}
+	return move(result);
+}
+
+static void ExternalQueryFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+	auto &data = (ExternalQueryFunctionData &)*data_p.bind_data;
+	if (data.finished) {
+		return;
+	}
+
+	if (data.rowiterator == nullptr) {
+		data.rowiterator = make_unique<QueryResultIterator>(data.tablerelation.get());
+	}
+
+	auto dconn = Connection(context.db->GetDatabase(context));
+
+	idx_t count = 0;
+
+	while ((data.rowiterator)->result != nullptr && count < STANDARD_VECTOR_SIZE) {
+		for (idx_t col = 0; col < data.tablerelation->types.size(); col++) {
+			output.SetValue(col, count, (*(*(data.rowiterator))).GetValue(col));
+		}
+		count++;
+		data.rowiterator->Next();
+	}
+	if ((data.rowiterator)->result == nullptr) {
+		data.finished = true;
+	}
+	// todo copy views
+	// {
+	// 	check_ok(sqlite3_prepare_v2(db, "SELECT sql FROM sqlite_master WHERE type='view'", -1, &res, nullptr), db);
+
+	// 	while (sqlite3_step(res) == SQLITE_ROW) {
+	// 		auto view_sql = string((const char *)sqlite3_column_text(res, 0));
+	// 		dconn.Query(view_sql);
+	// 	}
+	// 	check_ok(sqlite3_finalize(res), db);
+	// }
+	output.SetCardinality(count);
+}
+
+struct AttachFunctionData : public TableFunctionData {
+	AttachFunctionData() {
+	}
+
+	bool finished = false;
+	bool overwrite = false;
+	bool temporary = false;
+	string file_name = "";
+};
+
+static unique_ptr<FunctionData> AttachBind(ClientContext &context, TableFunctionBindInput &input,
+                                           vector<LogicalType> &return_types, vector<string> &names) {
+
+	auto result = make_unique<AttachFunctionData>();
+	result->file_name = input.inputs[0].GetValue<string>();
+
+	for (auto &kv : input.named_parameters) {
+		if (kv.first == "overwrite") {
+			result->overwrite = BooleanValue::Get(kv.second);
+		}
+		if (kv.first == "temporary") {
+			result->temporary = BooleanValue::Get(kv.second);
+		}
+	}
+
+	return_types.push_back(LogicalType::BOOLEAN);
+	names.emplace_back("Success");
+	return move(result);
+}
+
+static void AttachFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+	auto &data = (AttachFunctionData &)*data_p.bind_data;
+	if (data.finished) {
+		return;
+	}
+
+	DuckDB db {data.file_name};
+	auto econn = Connection(db);
+	auto dconn = Connection(context.db->GetDatabase(context));
+
+	{
+
+		auto q = econn.Query("select table_name from duckdb_tables();");
+		for (auto &row : *q) {
+			auto table_name = row.GetValue<string>(0);
+
+			auto table = econn.Table(table_name)->Execute();
+
+			vector<vector<Value>> table_values;
+			unique_ptr<DataChunk> trow = nullptr;
+			while ((trow = table->Fetch()) != nullptr) {
+				for (idx_t row_idx = 0; row_idx < trow->size(); row_idx++) {
+					vector<Value> row_values;
+					for (idx_t col = 0; col < table->names.size(); col++) {
+						row_values.push_back(trow->GetValue(col, row_idx));
```

Comment:
While this materialization should be removed entirely, I would just like to re-iterate that any use of the `Value` API is extremely slow. The `Value` type is intended to be used during planning and optimizing, not during execution.